migrations/educandu-2022-07-19-01-rename-question-to-text-in-interactive-media-plugin copy.js

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable camelcase, no-console, no-await-in-loop */
+export default class Educandu_2022_07_19_01_rename_question_to_text_in_interactive_media_plugin {
+  constructor(db) {
+    this.db = db;
+  }
+
+  async collectionUp(collectionName) {
+    const toUpdate = await this.db.collection(collectionName).find({ 'sections.type': 'interactive-media' }).toArray();
+    let updateCount = 0;
+
+    for (const doc of toUpdate) {
+      let docWasUpdated = false;
+      for (const section of doc.sections) {
+        if (section.type === 'interactive-media' && section.content) {
+          for (const chapter of section.content.chapters) {
+            chapter.text = chapter.question;
+            delete chapter.question;
+            docWasUpdated = true;
+          }
+          console.log(`Updating ${collectionName} ${doc._id} - section ${section.key}`);
+        }
+      }
+      if (docWasUpdated) {
+        updateCount += 1;
+        await this.db.collection(collectionName).replaceOne({ _id: doc._id }, doc);
+      }
+    }
+
+    return updateCount;
+  }
+
+  async collectionDown(collectionName) {
```

Comment:
Just as a note, a nice little trick for these mirror up/down methods would be to extract a method like `migrate(collectionName, oldProp, newProp)`, and with this we really ensure we didn't forget to update both methods when we find an issue in the `collectionUp` (which happened to me). We did that in `educandu-2022-07-11-01-rename-text-to-copyrightNotice-in-all-plugins`.

Reply:
I see. I did a similar thing. Maybe we should re-introduce the kind of migration helpers we once had (to process all documents in particular).